sessionctx/binloginfo/binloginfo.go

```diff
@@ -117,7 +166,12 @@ func (info *BinlogInfo) WriteBinlog(clusterID uint64) error {
 			logutil.BgLogger().Error("write binlog fail but error ignored")
 			metrics.CriticalErrorCounter.Add(1)
 			// If error happens once, we'll stop writing binlog.
-			atomic.CompareAndSwapUint32(&skipBinlog, skip, skip+1)
+			swapped := atomic.CompareAndSwapUint32(&skipBinlog, skip, skip+1)
+			if swapped && skip == 0 {
+				if err := statusListener(BinlogStatusSkipping); err != nil {
```

Comment:
It seems a TiDB server can only set the `BinlogStatus` to `BinlogStatusSkipping` during its life time. Is it expected?

Reply:
yes, we can't reload `binlog-enable` for now, the only chance of changing `BinlogStatus` is skipping binlog when binlog writing failed

Reply:
For the first time when the error is meet, the `skip` flag is set, and the `WriteBinlog` function is skipped since then.
Because after the first error, the data integrity is lost, writing binlogs is useless after failure recovery.